docs/transparent-proxy.md

```diff
@@ -0,0 +1,268 @@
+```
+
+**setup.sh**
+
+```shell
+#!/bin/bash -x
+#               enp0s3            enp0s8
+#  +--------+         \ +------+ /           +--------+
+#  | CLIENT | <-- 1 --> | MITM | <--- 2 ---> | SERVER |
+#  +--------+           +------+             +--------+
+#  10.13.37.10             br0              10.13.37.111
+
+# The IP of the RDP server which will receive proxied connections.
+SERVER_IP=10.13.37.111
+
+# The mark number to use in iptables (should be fine as-is)
+MARK=1
+
+# The routing table ID for custom rules (should be fine as-is)
+TABLE_ID=100
+
+# Create a custom routing table for pyrdp traffic
+echo "$TABLE_ID    pyrdp" >> /etc/iproute2/rt_tables
+
+# Intercept RDP traffic destined for the server (1)
+iptables -t mangle -I PREROUTING -p tcp -d 10.13.37.111 --dport 3389 \
+  -j TPROXY --tproxy-mark 0x1/0x1 --on-port 3389 --on-ip 127.0.0.1
+
+# Mark RDP return traffic intended for clients (2)
+iptables -t mangle -A PREROUTING \
+    -s $SERVER_IP \
+    -m tcp -p tcp --sport 3389 \
+    -j MARK --set-mark $MARK
+
+# Set route lookup to the pyrdp table for marked packets.
+ip rule add fwmark $MARK lookup $TABLE_ID
+
+# Add a custom route that redirects traffic intended for the outside world to loopback
+# So that server-client traffic passes through PyRDP
+# This table will only ever be used by RDP so it should not be problematic
+ip route add local default dev lo table $TABLE_ID
+
+# If you want the interception to happen at the Layer 2
+# Make sure that both your interfaces are bridged together
+
+# WARNING: if you have other important firewall rules, make sure to test the impact of this
+modprobe br_netfilter
+echo 1 > /proc/sys/net/bridge/bridge-nf-call-iptables
+
+# Target DROP in brouting means out of L2 go route in L3
+ebtables -t broute -A BROUTING -i enp0s3 -p ipv4 --ip-dst $SERVER_IP --ip-proto tcp --ip-dport 3389 -j redirect --redirect-target DROP
+ebtables -t broute -A BROUTING -i enp0s8 -p ipv4 --ip-src $SERVER_IP --ip-proto tcp --ip-source-port 3389 -j redirect --redirect-target DROP
+# recent linux like debian buster
+#ebtables-legacy -t broute -A BROUTING -i enp0s8 -p ipv4 --ip-src $SERVER_IP --ip-proto tcp --ip-source-port 3389 -j redirect --redirect-target DROP
+
+# Disable return path filtering
+echo 0 > /proc/sys/net/ipv4/conf/default/rp_filter
+echo 0 > /proc/sys/net/ipv4/conf/all/rp_filter
+echo 0 > /proc/sys/net/ipv4/conf/enp0s3/rp_filter
+echo 0 > /proc/sys/net/ipv4/conf/enp0s8/rp_filter
+
+# Make interception work without an IP assigned to br0
+# WARNING: unclear if required on not
+echo 1 > /proc/sys/net/ipv4/conf/br0/route_localnet
+echo 1 > /proc/sys/net/ipv4/conf/enp0s3/route_localnet
+
+# Mark socket Packets
+# WARNING: Unclear if required or not but likely good for performance so it will likely stay
+iptables -t mangle -N DIVERT
+iptables -t mangle -A DIVERT -j MARK --set-mark 1
+iptables -t mangle -A DIVERT -j ACCEPT
+iptables -t mangle -I PREROUTING -p tcp -m socket -j DIVERT
```

Comment:
I think I'll drop this. I'll test and drop if its not required.

Reply:
It wasn't required on my setup, but do confirm on your end as well.

Reply:
Dropped it